src/main/java/com/minecolonies/coremod/colony/buildings/modules/AbstractCraftingBuildingModule.java

```diff
@@ -67,8 +220,481 @@ public void serializeToView(PacketBuffer buf)
+                if (requiredItems.containsKey(itemStorage))
+                {
+                    amount += requiredItems.get(itemStorage).getA();
+                }
+                requiredItems.put(itemStorage, new Tuple<>(amount, false));
+            }
+
+            final ItemStorage output = new ItemStorage(recipeStorage.getA().getPrimaryOutput());
+            int amount = output.getAmount() * recipeStorage.getB();
+            if (requiredItems.containsKey(output))
+            {
+                amount += requiredItems.get(output).getA();
+            }
+            requiredItems.put(output, new Tuple<>(amount, false));
+        }
+
+        return new HashMap<>(requiredItems.entrySet()
+                               .stream()
+                               .collect(Collectors.toMap(key -> (stack -> stack.isItemEqualIgnoreDurability(key.getKey().getItemStack())), Map.Entry::getValue)));
+    }
+
+    @Override
+    public Map<ItemStorage, Integer> reservedStacks()
+    {
+        final Map<ItemStorage, Integer> recipeOutputs = new HashMap<>();
+        for (final Tuple<IRecipeStorage, Integer> recipeStorage : getPendingRequestQueue())
+        {
+            for (final ItemStorage itemStorage : recipeStorage.getA().getCleanedInput())
+            {
+                int amount = itemStorage.getAmount() * recipeStorage.getB();
+                if (recipeOutputs.containsKey(itemStorage))
+                {
+                    amount += recipeOutputs.get(itemStorage);
+                }
+                recipeOutputs.put(itemStorage, amount);
+            }
+        }
+        return recipeOutputs;
+    }
+
+    /**
+     * Get a list of all recipeStorages of the pending requests in the crafters queues.
+     * @return the list.
+     */
+    private List<Tuple<IRecipeStorage, Integer>> getPendingRequestQueue()
+    {
+        final List<Tuple<IRecipeStorage, Integer>> recipes = new ArrayList<>();
+        for (final ICitizenData citizen : building.getAssignedCitizen())
+        {
+            if (citizen.getJob() instanceof AbstractJobCrafter)
+            {
+                final List<IToken<?>> assignedTasks = new ArrayList<>(citizen.getJob(AbstractJobCrafter.class).getAssignedTasks());
+                assignedTasks.addAll(citizen.getJob(AbstractJobCrafter.class).getTaskQueue());
+
+                for (final IToken<?> taskToken : assignedTasks)
+                {
+                    final IRequest<? extends PublicCrafting> request = (IRequest<? extends PublicCrafting>) building.getColony().getRequestManager().getRequestForToken(taskToken);
+                    final IRecipeStorage recipeStorage = IColonyManager.getInstance().getRecipeManager().getRecipes().get(request.getRequest().getRecipeStorage());
+                    if (recipeStorage != null)
+                    {
+                        recipes.add(new Tuple<>(recipeStorage, request.getRequest().getCount()));

```

Comment:
If I'm understanding this correctly, it looks like it will count the recipes twice if there are two modules on the building, and so try to keep twice as many items as it should?  It doesn't look like there's any module-specific filter here.

Reply:
Good catch